modules/game_market/t_market.lua

```diff
@@ -0,0 +1,2512 @@
+  elseif widget:getId() == 'historyButton' then
+	marketHistoryBuy = {}
+	marketHistorySell = {}
+	sendMarketAction(1)
+  end
+
+  if widget:getId() ~= 'myOffers' and widget:getId() ~= 'historyButton' then
+	if lastItemID and lastItemID > 0 then
+		sendMarketAction(3, lastItemID, lastItemTier)
+	end
+	lastItemID = 0
+	lastItemTier = 0
+  end
+  
+  if marketMain:isVisible() then
+	if widget:getId() == 'historyButton' then
+		currentOffersPanel:setVisible(false)
+		offerHistoryPanel:setVisible(true)
+		marketWindow:setText(tr('Offer History'))
+	else
+		currentOffersPanel.sellSeparator:setVisible(false)
+		currentOffersPanel.sellStatusButton:setVisible(false)
+		currentOffersPanel.buySeparator:setVisible(false)
+		currentOffersPanel.buyStatusButton:setVisible(false)
+		currentOffersPanel.sellEndButton:setWidth(220)
+		currentOffersPanel.buyEndButton:setWidth(220)
+		currentOffersPanel:setVisible(true)
+		offerHistoryPanel:setVisible(false)
+		marketWindow:setText(tr('My Offers'))
+	end
+	marketMain:setVisible(false)
+	closeButton:setVisible(false)
+	marketHistory:setVisible(true)
+  elseif marketHistory:isVisible() then
+	if widget:getId() == 'historyButton' then
+		currentOffersPanel:setVisible(false)
+		offerHistoryPanel:setVisible(true)
+		marketWindow:setText(tr('Offer History'))
+	elseif widget:getId() == 'myOffers' or widget:getId() == 'currentOffers' then
+		if currentOffersPanel.sellSeparator then
+			currentOffersPanel.sellSeparator:setVisible(false)
+		end
+		if currentOffersPanel.sellStatusButton then
+			currentOffersPanel.sellStatusButton:setVisible(false)
+		end
+		if currentOffersPanel.buySeparator then
+			currentOffersPanel.buySeparator:setVisible(false)
+		end
+		if currentOffersPanel.buyStatusButton then
+			currentOffersPanel.buyStatusButton:setVisible(false)
+		end
+		if currentOffersPanel.sellEndButton then
+			currentOffersPanel.sellEndButton:setWidth(220)
+		end
+		if currentOffersPanel.buyEndButton then
+			currentOffersPanel.buyEndButton:setWidth(220)
+		end
+		offerHistoryPanel:setVisible(false)
+		currentOffersPanel:setVisible(true)
+		marketWindow:setText(tr('My Offers'))
+	elseif widget:getId() == 'marketButton' then
+		lastSelectedMySell = nil
+		lastSelectedMyBuy = nil
+		lastSelectedHistorySell = nil
+		lastSelectedHistoryBuy = nil
+		marketHistory:setVisible(false)
+		marketMain:setVisible(true)
+		detailsMarket:setVisible(false)
+		marketPanel:setVisible(true)
+		closeButton:setVisible(true)
+		marketWindow:setText(tr('Market'))
+	end
+  else
+	lastSelectedMySell = nil
+	lastSelectedMyBuy = nil
+	lastSelectedHistorySell = nil
+	lastSelectedHistoryBuy = nil
+	marketHistory:setVisible(false)
+	marketMain:setVisible(true)
+	detailsMarket:setVisible(false)
+	marketPanel:setVisible(true)
+	closeButton:setVisible(true)
+	marketWindow:setText(tr('Market'))
+  end
+end
+
+function getDepotItemCount(itemId, tier)
+	for _, data in pairs(depotLockerItems) do
+		if data[1] == itemId and data[2] == tier then
```

Comment:
```suggestion
		if data[1] == itemId and data[2] == tier then
```
By programming logic, if you put **AND tier.**

By logic, you will break all old protocols that do not have `g_game.getFeature(Otc::GameThingUpgradeClassification)` (untest)

 check `void ProtocolGame::parseMarketEnterOld(const InputMessagePtr& msg)`
 not have tier var
 
```diff
     for (auto i = 0; i < itemsSent; ++i) {
        const uint16_t itemId = msg->getU16();
        const uint16_t count = msg->getU16();
+        const uint8_t itemTier = 0;
-       depotItems.push_back({ itemId, count });
+       depotItems.push_back({ itemId, itemTier, count });
    }
```

Reply:
Fixed, thanks for the comment.